pkg/controllers/nsautoprop/controller.go

```diff
@@ -73,162 +70,179 @@ Note that since both NamespaceAutoPropagationController and global-scheduler set
+			logger.Error(err, "Failed to list all namespaces")
+			return
+		}
+
+		for _, ns := range allNamespaces {
+			c.worker.Enqueue(common.QualifiedName{Name: naming.GenerateFederatedObjectName(ns.Name, c.typeConfig.Name)})
 		}
 	}
-	c.clusterInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+
+	if _, err := c.clusterInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			reconcileAll()
 		},
 		DeleteFunc: func(obj interface{}) {
 			reconcileAll()
 		},
-	}, util.NoResyncPeriod)
+	}, util.NoResyncPeriod); err != nil {
+		return nil, err
+	}
+
 	return c, nil
 }
 
-func (c *Controller) reconcile(qualifiedName common.QualifiedName) worker.Result {
-	key := qualifiedName.String()
+func (c *Controller) reconcile(ctx context.Context, qualifiedName common.QualifiedName) worker.Result {
+	keyedLogger := c.logger.WithValues("federated-name", qualifiedName.String())
+	ctx = klog.NewContext(ctx, keyedLogger)
 
 	c.metrics.Rate("namespace-auto-propagation-controller.throughput", 1)
-	klog.V(4).Infof("namespace auto propagation controller starting to reconcile %v", key)
+	keyedLogger.V(3).Info("Starting to reconcile")
 	startTime := time.Now()
 	defer func() {
 		c.metrics.Duration("namespace-auto-propagation-controller.latency", startTime)
-		klog.V(4).
-			Infof("namespace auto propagation controller finished reconciling %v (duration: %v)", key, time.Since(startTime))
+		keyedLogger.WithValues("duration", time.Since(startTime)).V(3).Info("Finished reconciling")
 	}()
 
-	fedNamespace, err := c.getFederatedObject(qualifiedName)
-	if err != nil {
-		utilruntime.HandleError(err)
+	fedNamespace, err := c.clusterFedObjectInformer.Lister().Get(qualifiedName.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		keyedLogger.Error(err, "Failed to get federated namespace")
 		return worker.StatusError
 	}
-	if fedNamespace == nil || fedNamespace.GetDeletionTimestamp() != nil {
+
+	if apierrors.IsNotFound(err) || fedNamespace.GetDeletionTimestamp() != nil {
 		return worker.StatusAllOK
 	}
 
 	if ok, err := pendingcontrollers.ControllerDependenciesFulfilled(
 		fedNamespace,
 		PrefixedNamespaceAutoPropagationControllerName,
 	); err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to check controller dependencies for %q: %w", key, err))
+		keyedLogger.Error(err, "Failed to get pending controllers")
 		return worker.StatusError
 	} else if !ok {
 		return worker.StatusAllOK
 	}
 
-	if !c.shouldBeAutoPropagated(fedNamespace) {
+	srcMeta, err := fedNamespace.Spec.GetTemplateAsUnstructured()
+	if err != nil {
+		keyedLogger.Error(err, "Failed to get source object's metadata from ClusterFederatedObject")
+		return worker.StatusError
+	}
+
+	if !c.shouldBeAutoPropagated(srcMeta) {
 		updated, err := pendingcontrollers.UpdatePendingControllers(
 			fedNamespace,
 			PrefixedNamespaceAutoPropagationControllerName,
 			false,
 			c.typeConfig.GetControllers(),
```

Comment:
The current implementation doesn't support the namespace FTC being updated, is this acceptable? cc @gary-lgy